tests/orquestra/quantum/wip/operators/_operators_test.py

```diff
@@ -4,149 +4,286 @@
-        assert "Badly formatted" in str(e.value)
+            PauliTerm(pauli_str)
 
-        # Test passing invalid qubit number
-        with pytest.raises(ValueError) as e:
-            PauliTerm("X-1254")
-        assert "Badly formatted" in str(e.value)
-
-        # Test passing wrong operator
-        with pytest.raises(ValueError) as e:
-            PauliTerm("A0")
         assert "Badly formatted" in str(e.value)
 
-    def test_pauliterm_from_list_constructor(self, pauli_term):
-        assert len(pauli_term._ops) == 3
-        assert pauli_term[0] == "X"
-        assert pauli_term[1] == "Y"
-        assert pauli_term[12] == "Z"
+    @pytest.mark.parametrize("operator_dict", [{0: "XX"}, {0: "X", 1: "A"}])
+    def test_term_cannot_be_constructed_from_dictionary_containing_incorrect_operators(
+        self, operator_dict
+    ):
+        with pytest.raises(ValueError):
+            PauliTerm(operator_dict)
 
-        # Error for wrong tuple shape
+    @pytest.mark.parametrize("pauli_str", ["X0 * Y0", "X0 * Y1 * Z1"])
+    def test_term_cannot_be_constructed_if_duplicate_qubits_are_present_in_pauli_str(
+        self, pauli_str
+    ):
+        with pytest.raises(ValueError):
+            PauliTerm(pauli_str)
+
+    @pytest.mark.parametrize(
+        "pauli_str, coefficient", [("0.5 * X0", 0.5), ("2.0 * Y12", 3.0)]
+    )
+    def test_term_cannot_be_constructed_if_coefficient_is_passed_via_both_ways(
+        self, pauli_str, coefficient
+    ):
+        with pytest.raises(ValueError) as e:
+            PauliTerm(pauli_str, coefficient)
+
+        assert "Coefficient can be provided either in an argument " in str(e.value)
+
+    @pytest.mark.parametrize(
+        "term",
+        [
+            PauliTerm("X0", -5.0),
+            PauliTerm("-5.0 * Z1"),
+            PauliTerm("2.0 * X0 * Y2 * Z3"),
+            PauliTerm("X0 * Y2 * Z3", 2.0),
+        ],
+    )
+    def test_creating_term_from_its_str_representation_gives_the_same_term(self, term):
+        assert PauliTerm(str(term)) == term
+
+
+class TestConstructingPauliTermFromList:
+    @pytest.mark.parametrize(
+        "op_list, coefficient",
+        [
+            ([("X", 0), ("Y", 1), ("Z", 12)], -0.5),
+        ],
+    )
+    def test_term_constructed_from_list_has_expected_operators_and_coefficient(
+        self, op_list, coefficient
+    ):
+        term = PauliTerm.from_list(op_list, coefficient)
+        # The reverse (pair[::-1]) is from the fact, that for whatever reason
+        # from_list accepts input in the reverse direction then the
+        # operations_as_set produces it.
+        assert term.operations_as_set() == frozenset([pair[::-1] for pair in op_list])
+        assert term.coefficient == coefficient
+
+    def test_term_cannot_be_constructed_from_list_of_incorrectly_shaped_tuples(self):
         with pytest.raises(ValueError) as e:
             PauliTerm.from_list([("X0")])
+
         assert "PauliTerm.from_str" in str(e.value)
```

Comment:
this error message should be changed as we are no longer using the `PauliTerm.from_str` method

Reply:
Changed in [6e4fac1](https://github.com/zapatacomputing/orquestra-quantum/pull/44/commits/6e4fac19fc25b642d4e66f23c45fc31e9e18fc03)